### PR TITLE
[Autoround] add custom dataset support

### PR DESCRIPTION
**Pull request by yiliu30**
```bash
vllm ({'pretrained': 'Qwen3-8B-W4A16-G128-AutoRound-Ultrachat200k/', 'tensor_parallel_size': 1, 'max_model_len': 8192, 'max_num_batched_tokens': 32768, 'max_num_seqs': 128, 'add_bos_token': True, 'gpu_memory_utilization': 0.8, 'dtype': 'bfloat16', 'max_gen_toks': 2048, 'enable_prefix_caching': False}), gen_kwargs: ({}), limit: None, num_fewshot: None, batch_size: 128
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8832|±  |0.0088|
|     |       |strict-match    |     5|exact_match|↑  |0.8832|±  |0.0088|
```

https://huggingface.co/Yi30/Qwen3-8B-W4A16-G128-AutoRound-Ultrachat200k

TEST PLAN:
"please outline how the changes were tested"

### CODE REVIEW

File: examples/autoround/quantization_w4a16/qwen3_example_custom_dataset.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from datasets import load_dataset
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from llmcompressor import oneshot
+from llmcompressor.modifiers.autoround import AutoRoundModifier, fix_batch_if_needed
+from llmcompressor.utils import dispatch_for_generation
+
+MODEL_ID = "Qwen/Qwen3-8B"
+DATASET_ID = "HuggingFaceH4/ultrachat_200k"
+DATASET_SPLIT = "train_sft"
```
[the remark this answers is not in the document]
Comment (yiliu30) [reply examples/autoround/quantization_w4a16/qwen3_example_custom_dataset.py:10]: Done

File: examples/autoround/quantization_w4a16/qwen3_example_custom_dataset.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+NUM_CALIBRATION_SAMPLES = 128
+MAX_SEQUENCE_LENGTH = 2048
+ITERS = 200
+model = AutoModelForCausalLM.from_pretrained(MODEL_ID, torch_dtype="auto")
+tokenizer = AutoTokenizer.from_pretrained(MODEL_ID)
+tokenizer.pad_token = tokenizer.eos_token
```
[the remark this answers is not in the document]
Comment (yiliu30) [reply examples/autoround/quantization_w4a16/qwen3_example_custom_dataset.py]: It will automatically onload to the GPU.

File: src/llmcompressor/modifiers/autoround/utils.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    if torch.all(mask == 0):
+        raise ValueError(
+            "Invalid causal attention mask for AutoRound: all positions are masked"
+        )
+
+    global_min = mask.amin()
+    return (mask.amax(dim=-1) > global_min).to(torch.long)
+
+
+def fix_attention_mask(
```
Comment (brian-dellabetta) [inline src/llmcompressor/modifiers/autoround/utils.py:34]: Hi @yiliu30 , can you elaborate on this? Am i understanding that this the majority of changes you want to add and this is specific to a detail to autoround and not something we'd want in higher scope? like in data-loading modules?

Comment (yiliu30) [reply src/llmcompressor/modifiers/autoround/utils.py:34]: Hi @brian-dellabetta, yes, this is AutoRound-specific behavior for calculating the loss on calibration samples with different lengths.

Please check the code here — I also added a note in the comments.
https://github.com/intel/auto-round/blob/50ee58c9e176e9da2a744dbe6ed220f26e80eccd/auto_round/calibration/llm.py#L315-L355

Comment (brian-dellabetta) [reply src/llmcompressor/modifiers/autoround/utils.py:34]: got it, thanks @yiliu30

**Review by brian-dellabetta — APPROVED** [review]
